services/viva/helpers/populateForm.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  getFormFieldsWithLoadPreviousAttribute,
+  formatAnswer,
+  mergeAnswers,
+} from '../libs/formAnswers';
+
+import type { CaseForm, CaseFormAnswer, CasePerson } from '../types/caseItem';
+
+export interface FormField {
+  id: string;
+  loadPrevious: string[];
+  tags: string[];
+  type: string;
+}
+
+export function populateChildrenAnswers(
+  repeaterInputList: FormField[],
+  childrenList: CasePerson[]
+): CaseFormAnswer[] {
+  const answers = childrenList.reduce((answers, child, index) => {
+    const childAnswers = Object.keys(child)
+      .map(key => {
+        const childAnswerList = repeaterInputList
+          .map(input => {
+            if (input.tags.includes(key)) {
+              const inputId = input.id.replace('[*]', index.toString());
+              return formatAnswer(
+                inputId,
+                input.tags,
+                child[key as keyof CasePerson]
+              ) as CaseFormAnswer;
+            }
+            return undefined;
+          })
+          .filter(Boolean) as CaseFormAnswer[];
+
+        const [childAnswer] = childAnswerList;
+        return childAnswer;
+      })
+      .filter(Boolean);
+
+    return [...answers, ...childAnswers];
+  }, [] as CaseFormAnswer[]);
+
+  return answers;
```

Comment:
This block of code is very hard to read. One solution would be to split every callback function used in .map to a separate function

Reply:
will fix in later PR